src/System.Security.Principal.Windows/src/System/Security/Principal/WindowsIdentity.cs

```diff
@@ -208,20 +217,72 @@ public WindowsIdentity(string sUserPrincipalName)
 
                             if (profileBuffer != null)
                                 profileBuffer.Dispose();
-
-                            _safeTokenHandle = accessTokenHandle;
                         }
                     }
                 }
             }
         }
 
-        private static SafeLsaHandle ConnectToLsa()
+        /// <summary>
+        /// To obtain SafeLsaHandle we try before for an Impersonation token that require SeTcbPrivilege, if fails we fallback
+        /// to an Identification token
+        /// </summary>
+        /// <param name="logonProcessNameBuffer"></param>
+        /// <returns></returns>
+        private static SafeLsaHandle GetLsaHandle(byte[] logonProcessNameBuffer)
```

Comment:
If there's only one caller to this method is it worth taking the parameter, or should it just be a static field reference within the method?